frontend/src/framework/internal/components/RightSettingsPanel/private-components/RealizationFilterSettings/realizationFilterSettings.tsx

```diff
@@ -77,19 +95,108 @@ export const RealizationFilterSettings: React.FC<RealizationFilterSettingsProps>
         }
         setEnsembleIdentStringHasUnsavedChangesMap(updatedHasUnsavedChangesMap);
         setEnsembleIdentStringToRealizationFilterSelectionsMap(updatedSelectionsMap);
-    }
 
-    function handleFilterSettingsClose() {
-        // Check if there are unsaved changes
-        const hasUnsavedChanges = Object.values(ensembleIdentStringHasUnsavedChangesMap).some(
+        // Update number of unsaved filters
+        const numberOfUnsavedFilters = Object.values(updatedHasUnsavedChangesMap).filter(
             (hasUnsavedChanges) => hasUnsavedChanges
-        );
-        if (hasUnsavedChanges) {
-            setDialogOpen(true);
-        } else {
-            props.onClose();
-            setActiveFilterEnsembleIdent(null);
+        ).length;
+        setNumberOfUnsavedRealizationFilters(numberOfUnsavedFilters);
+    }
+
+    const handleApplyAllClick = React.useCallback(
+        function handleApplyAllClick() {
+            // Apply all the unsaved changes state and reset the unsaved changes state
+            const resetHasUnsavedChangesMap: { [ensembleIdentString: string]: boolean } = {};
+            for (const ensembleIdentString of Object.keys(ensembleIdentStringToRealizationFilterSelectionsMap)) {
```

Comment:
I you want to iterate the keys, you can simplify this by using a [`for...in` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in)